WordPress/src/test/java/org/wordpress/android/util/crashlogging/WPCrashLoggingDataProviderTest.kt

```diff
@@ -0,0 +1,179 @@
+import com.nhaarman.mockitokotlin2.whenever
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.SoftAssertions
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.junit.MockitoJUnitRunner
+import org.wordpress.android.R
+import org.wordpress.android.fluxc.model.AccountModel
+import org.wordpress.android.fluxc.store.AccountStore
+import org.wordpress.android.util.BuildConfigWrapper
+import org.wordpress.android.util.EncryptedLogging
+import org.wordpress.android.util.LocaleManagerWrapper
+import org.wordpress.android.util.LogFileProviderWrapper
+import org.wordpress.android.util.crashlogging.WPCrashLoggingDataProvider.Companion.EVENT_BUS_EXCEPTION
+import org.wordpress.android.util.crashlogging.WPCrashLoggingDataProvider.Companion.EVENT_BUS_INVOKING_SUBSCRIBER_FAILED_ERROR
+import org.wordpress.android.util.crashlogging.WPCrashLoggingDataProvider.Companion.EVENT_BUS_MODULE
+import org.wordpress.android.util.crashlogging.WPCrashLoggingDataProvider.Companion.EXTRA_UUID
+import org.wordpress.android.viewmodel.ResourceProvider
+import java.io.File
+import java.util.Locale
+
+@RunWith(MockitoJUnitRunner::class)
+class WPCrashLoggingDataProviderTest {
+    lateinit var sut: WPCrashLoggingDataProvider
+
+    private val mockedFile: File = mock { on { exists() } doReturn true }
+
+    private val logFileProvider: LogFileProviderWrapper = mock {
+        on { getLogFiles() } doReturn listOf(mockedFile)
+    }
+
+    private val encryptedLogging: EncryptedLogging = mock {
+        on { encryptAndUploadLogFile(any(), any()) } doReturn TEST_UUID
+    }
+
+    private val resourceProvider: ResourceProvider = mock {
+        on { getString(R.string.pref_key_send_crash) } doReturn SEND_CRASH_SAMPLE_KEY
+    }
+
+    private val accountStore: AccountStore = mock()
+    private val localeManager: LocaleManagerWrapper = mock()
+    private val buildConfig: BuildConfigWrapper = mock()
+    private val sharedPreferences: SharedPreferences = mock()
+
+    @Before
+    fun setUp() {
+        sut = WPCrashLoggingDataProvider(
+                sharedPreferences,
+                resourceProvider,
+                accountStore,
+                localeManager,
+                encryptedLogging,
+                logFileProvider,
+                buildConfig
+        )
+    }
+
+    @Test
+    fun `should contain encrypted logs key for extra known keys`() {
+        assertThat(sut.extraKnownKeys()).contains(EXTRA_UUID)
+    }
+
+    @Test
+    fun `should correctly reads and maps user`() {
+        whenever(accountStore.account).thenReturn(TEST_ACCOUNT)
+
+        val user = sut.userProvider()
+
+        SoftAssertions().apply {
+            assertThat(user?.username).isEqualTo(TEST_ACCOUNT.userName)
+            assertThat(user?.email).isEqualTo(TEST_ACCOUNT.email)
+            assertThat(user?.userID).isEqualTo(TEST_ACCOUNT.userId.toString())
+        }.assertAll()
```

Comment:
TIL about `SoftAssertions`! I used to do something similar using Truth, but I didn't know this was possible with AssertJ.

Reply:
True, `SoftAssertion` is a cool idea, I really like it!

I just remembered - it's quite easy to forget about calling `assertAll()` at the end, when using Kotlin's `apply`. In 7c7b1a32f12a10d168e34ccf67f958729e79e495 I've moved to use `assertSoftly` which is a static function provided by AssertJ that does the same but with no need to call `assertAll()`.